presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGenerator.java

```diff
@@ -515,6 +517,10 @@ public PinotQueryGeneratorContext visitTopN(TopNNode node, PinotQueryGeneratorCo
             requireNonNull(context, "context is null");
             checkSupported(!forbidBrokerQueries, "Cannot push topn in segment mode");
             checkSupported(node.getStep().equals(TopNNode.Step.SINGLE), "Can only push single logical topn in");
+
+            if (!pushdownTopnBrokerQueries) {
```

Comment:
shall we throw exception here?
am expecting:

if (pushdownTopnBrokerQueries) {
    return context.withTopN(...)
}
return context;

Reply:
Throwing an exception is indeed the right thing to do here: The whole Pinot pushdown framework uses exceptions to know "when to push no further". The exception will be propagated all the way to PinotPushdownPlanOptimizer and will stall future pushdown. The exception trick is used across the entire file and so I would vote for consistency with the rest of the file.

Reply:
get it.